src/python/WMCore/WMStats/CherryPyThreads/BuildParentLock.py

```diff
@@ -28,14 +30,19 @@ def setConcurrentTasks(self, config):
     def fetchIncludeParentsRequests(self, config):
         """
         Fetch active requests from the DataCache that have IncludeParents=True
+        and send to reqmgr2 auxiliary database.
         """
         # use this boolean to signal whether there were datasets that failed
         # to get their parentage resolved
         incompleteParentage = False
 
         setDsets = set()
         setParents = set()
+        dictParents = {}
         self.logger.info("Executing parent lock cherrypy thread")
+        if DataCache.isEmpty():
```

Comment:
When reqmon starts, the data cache is not populated by the time the build parent lock thread runs. If the cache is empty, the thread will overwrite whatever is in the aux db with a blank list. This is bad. Bailing out of the thread if the cache is empty can prevent this, but there might be another way.